src/components/inline_edit/inline_edit_title.test.tsx

```diff
@@ -10,21 +10,65 @@ import React from 'react';
 import { render } from '../../test/rtl';
 import { requiredProps } from '../../test/required_props';
 
-import { EuiInlineEditTitle } from './inline_edit_title';
+import {
+  EuiInlineEditTitle,
+  EuiInlineEditTitleProps,
+} from './inline_edit_title';
+import { TITLE_SIZES } from '../title/title';
 
 describe('EuiInlineEditTitle', () => {
-  describe('props', () => {
-    test('renders as title', () => {
-      const { container } = render(
-        <EuiInlineEditTitle
-          inputAriaLabel={'textInput'}
-          defaultValue="hello world"
-          heading="h2"
-          {...requiredProps}
-        />
-      );
-
-      expect(container.firstChild).toMatchSnapshot();
+  const inlineEditTitleProps: EuiInlineEditTitleProps = {
+    ...requiredProps,
+    inputAriaLabel: 'Edit title inline',
+    defaultValue: 'Hello World!',
+    heading: 'h1',
+  };
+
+  it('renders in readMode', () => {
+    const { container, getByTestSubject, getByText } = render(
+      <EuiInlineEditTitle {...inlineEditTitleProps} />
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+
+    expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+    expect(getByText('Hello World!')).toBeTruthy();
+  });
+
+  it('renders in editMode', () => {
+    const { container, getByTestSubject, queryByTestSubject } = render(
+      <EuiInlineEditTitle {...inlineEditTitleProps} startWithEditOpen={true} />
+    );
+
+    expect(container.firstChild).toMatchSnapshot();
+
+    expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
+    expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
+  });
+
+  describe('Title Sizes', () => {
+    TITLE_SIZES.forEach((size) => {
+      it(`renders in size ${size} in readMode`, () => {
+        const { container } = render(
+          <EuiInlineEditTitle {...inlineEditTitleProps} size={size} />
+        );
+
+        expect(container.firstChild).toMatchSnapshot();
+      });
+    });
+
+    TITLE_SIZES.forEach((size) => {
+      it(`renders in size ${size} in readMode`, () => {
+        const { container } = render(
+          <EuiInlineEditTitle
+            {...inlineEditTitleProps}
+            startWithEditOpen={true}
+            size={size}
+          />
+        );
+
+        expect(container.firstChild).toMatchSnapshot();
+      });
```

Comment:
see above. there's no need to test code that isn't in this file

```suggestion
    TITLE_SIZES.forEach((size) => {
      it(`renders size ${size}`, () => {
        const { container } = render(
          <EuiInlineEditTitle {...inlineEditTitleProps} size={size} />
        );

        expect(container.firstChild).toMatchSnapshot();
      });
```